package/gluon-core/files/lib/netifd/proto/gluon_wired.sh

```diff
@@ -7,15 +7,29 @@ init_proto "$@"
 proto_gluon_wired_init_config() {
         proto_config_add_boolean transitive
         proto_config_add_int index
+        proto_config_add_boolean add_vxlan_layer
+        proto_config_add_string vxpeer6addr
 }
 
 xor2() {
         echo -n "${1:0:1}"
         echo -n "${1:1:1}" | tr '0123456789abcdef' '23016745ab89efcd'
 }
 
+is_layer3_device () {
+        local addrlen="$(cat "/sys/class/net/$1/addr_len")"
+        test "$addrlen" -eq 0
+}
+
 # shellcheck disable=SC2086
 interface_linklocal() {
+        if is_layer3_device "$1"; then
+                ubus call network.interface dump | \
+                        jsonfilter -e "@.interface[@.l3_device='$1']['ipv6-address'][*].address" | \
+                        grep -e '^fe[89ab]' | head -n 1
```

Comment:
Hmm, I assume this is about L3 VPN tunnels like Wireguard? How will link-local addresses be assigned for that?

Maybe we should enforce the configuration of an explicit `vxip6addr` in addition to `vxpeer6addr` in such cases, and only support L2 interfaces when no local address is configured.

Reply:
It's dynamically added via `proto_add_ipv6_address` in a netifd script.

Reply:
Please let me know when you have decided if you want to introduce `vxip6addr` instead.

Reply:
I think using the LL address of the underlying interface is fine for now, but the code will need error handling to make the setup fail when no such address exists.